pkg/destroy/azure/azure.go

```diff
@@ -100,26 +101,85 @@ func New(logger logrus.FieldLogger, metadata *types.ClusterMetadata) (providers.
 }
 
 // Run is the entrypoint to start the uninstall process.
-func (o *ClusterUninstaller) Run() error {
+func (o *ClusterUninstaller) Run() (err error) {
+	var errs []error
+
 	o.configureClients()
 
-	o.Logger.Debug("deleting public records")
-	if err := deletePublicRecords(context.TODO(), o.zonesClient, o.recordsClient, o.privateZonesClient, o.privateRecordSetsClient, o.Logger, o.ResourceGroupName); err != nil {
+	group := o.InfraID + "-rg"
+	timeout := 30 * time.Minute
+
+	ctx, cancel := context.WithTimeout(context.Background(), timeout)
+	defer cancel()
+
+	wait.Until(
```

Comment:
https://godoc.org/k8s.io/apimachinery/pkg/util/wait#UntilWithContext should be better with the context?

Reply:
Fixed.